widgets/give_form.php

```diff
@@ -1,259 +1,396 @@
+            // is this v3 form?
+            if (get_post_meta($form_id, 'formBuilderSettings', true)) {
+                if ($donationForm = DonationForm::find($form_id)) {
+                    $donationForm->settings->showHeading        = boolval($show_title);
+                    $donationForm->settings->enableDonationGoal = boolval($show_goal);
+                    $donationForm->save();
+                }
+            }
+        }
+
+        $shortcode = sprintf(
+            '[give_form id="%s" show_title="%s" show_goal="%s" show_content="%s" display_style="%s" continue_button_title="%s"]',
+            $form_id,
+            $show_title,
+            $show_goal,
+            $show_content,
+            $display_style,
+            $continue_button_title
+        );
+
+        echo '<div class="givewp-elementor-widget give-form-shortcode-wrap">';
+
+        echo do_shortcode($shortcode);
+
+        echo '</div>';
+    }
+
+    /**
+     * @return array
+     */
+    private function get_donation_forms_options()
+    {
+        $options = [];
+
+        $forms = DB::table('posts')
+                   ->select('ID', 'post_title')
+                   ->where('post_type', 'give_forms')
+                   ->where('post_status', 'publish')
+                   ->getAll();
+
+        foreach ($forms as $form) {
+            $options[$form->ID] = $form->post_title;
+        }
+
+        return $options;
+    }
+
+    /**
+     * Get forms using legacy template from list of forms returned by DW4Elementor_GiveWP_Form_Widget::get_donation_forms_options
+     *
+     * @unlreased
+     *
+     * @param array $forms
+     *
+     * @return array
+     */
+    private function get_legacy_forms($forms)
+    {
+        $data = [];
+
+        foreach (array_keys($forms) as $formId) {
+            if ('legacy' === $this->get_form_template($formId)) {
+                $data[] = (string)$formId;
+            }
+        }
+
+        return $data;
+    }
+
+    /**
+     * Get forms using classic template from list of forms returned by DW4Elementor_GiveWP_Form_Widget::get_donation_forms_options
+     *
+     * @unlreased
+     *
+     * @param array $forms
+     *
+     * @return array
+     */
+    private function get_classic_forms($forms)
+    {
+        $data = [];
+
+        foreach (array_keys($forms) as $formId) {
+            if ('classic' === $this->get_form_template($formId)) {
+                $data[] = (string)$formId;
+            }
+        }
+
+        return $data;
+    }
+
+    /**
+     * Get v3 forms from list of forms returned by DW4Elementor_GiveWP_Form_Widget::get_donation_forms_options
+     *
+     * @unlreased
+     *
+     * @param array $forms
+     *
+     * @return array
+     */
+    private function get_v3_forms($forms)
+    {
+        $data = [];
+
+        foreach (array_keys($forms) as $formId) {
+            if (get_post_meta($formId, 'formBuilderSettings', true)) {
```

Comment:
One little change here, let's use the helper `Give\Helpers\Form::isV3Form($int)` (only framework helper we have at the moment) -  in case we decide to change the storage in the future -    we can abstract away the actual logic for determination

Reply:
@jonwaldstein resolved https://github.com/impress-org/givewp-elementor-widgets/pull/19/commits/d06870ab5bc5f6febfb34df43552844f55e99eac

Reply:
thank!